thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java

```diff
@@ -343,14 +343,21 @@ void updateDetectionPipeline(long detectionID, String yamlDetectionConfig, long
     // Translate config from YAML to detection config (JSON)
     TreeMap<String, Object> newDetectionConfigMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
     newDetectionConfigMap.putAll(ConfigUtils.getMap(this.yaml.load(yamlDetectionConfig)));
-    detectionConfig = buildDetectionConfigFromYaml(startTime, endTime, newDetectionConfigMap, existingDetectionConfig);
-    detectionConfig.setYaml(yamlDetectionConfig);
-
-    // Validate updated config before saving it
-    detectionValidator.validateUpdatedConfig(detectionConfig, existingDetectionConfig);
-    // Save the detection config
-    Long id = this.detectionConfigDAO.save(detectionConfig);
-    Preconditions.checkNotNull(id, "Error while saving the detection pipeline");
+    // If it is to disable the pipeline then no need to do validation and parsing.
+    // It is possible that the metric or dataset was deleted so the validation will fail.
+    if (!MapUtils.getBooleanValue(newDetectionConfigMap, PROP_ACTIVE, true)) {
+      existingDetectionConfig.setActive(false);
```

Comment:
The raw yaml config needs to be injected into the detection object before saving. Otherwise, the config will still show `active: true`.

Reply:
Good point. I put it into "finally" now. Do you have other ideas?